File: packages/typescript/examples/project-big-five-to-abbf.ts
import {
  AdaptiveBifurcatedProfile,
  BigFiveAgreeableness,
  BigFiveConscientiousness,
  BigFiveExtraversion,
  BigFiveNeuroticism,
  BigFiveOpenness,
  BigFiveTraits,
} from "../dist/index.js";

const traits = new BigFiveTraits({
  openness: new BigFiveOpenness({
    aestheticSensitivityScore: 0.75,
    creativeImaginationScore: 0.7,
    intellectualCuriosityScore: 0.8,
  }),
  conscientiousness: new BigFiveConscientiousness({
    organizationScore: 0.65,
    responsibilityScore: 0.7,
    productivityScore: 0.75,
  }),
  extraversion: new BigFiveExtraversion({
    assertivenessScore: 0.4,
    sociabilityScore: 0.45,
    energyLevelScore: 0.5,
  }),
  agreeableness: new BigFiveAgreeableness({
    compassionScore: 0.8,
    respectfulnessScore: 0.75,
    trustScore: 0.7,
  }),
  neuroticism: new BigFiveNeuroticism({
    anxietyScore: 0.25,
    emotionalVolatilityScore: 0.3,
    depressionScore: 0.2,
  }),
});

const profile = AdaptiveBifurcatedProfile.fromBigFive(traits);
for (const axis of profile.axes) {
  console.log(
    `${axis.domain}: ${axis.score.toFixed(2)} (${axis.dominantPole ?? "balanced"})`,
  );
}

File: packages/python/examples/project_big_five_to_abbf.py
"""Project fixed Big Five traits into an ABBF vector."""

from __future__ import annotations

from personalitygen import (
    AdaptiveBifurcatedProfile,
    BigFiveAgreeableness,
    BigFiveConscientiousness,
    BigFiveExtraversion,
    BigFiveNeuroticism,
    BigFiveOpenness,
    BigFiveTraits,
)


def main() -> None:
    traits = BigFiveTraits(
        openness=BigFiveOpenness(0.75, 0.70, 0.80),
        conscientiousness=BigFiveConscientiousness(0.65, 0.70, 0.75),
        extraversion=BigFiveExtraversion(0.40, 0.45, 0.50),
        agreeableness=BigFiveAgreeableness(0.80, 0.75, 0.70),
        neuroticism=BigFiveNeuroticism(0.25, 0.30, 0.20),
    )
    profile = AdaptiveBifurcatedProfile.from_big_five(traits)

    print("Projected ABBF axes:")
    for axis in profile.axes:
        pole = axis.dominant_pole.value if axis.dominant_pole else "balanced"
        print(f"{axis.domain.value}: {axis.score:.2f} ({pole})")


if __name__ == "__main__":
    main()
